opm/simulators/flow/RFTContainer.cpp

```diff
@@ -0,0 +1,225 @@
+           const Parallel::Communication& comm)
+{
+    const auto& rft_config = schedule_[reportStepNum].rft_config();
+    if (!rft_config.active()) {
+        return;
+    }
+
+    if (comm.size() > 1) {
+        gatherAndUpdateMap(oilConnectionPressures_, comm);
+        gatherAndUpdateMap(waterConnectionSaturations_, comm);
+        gatherAndUpdateMap(gasConnectionSaturations_, comm);
+    }
+
+    for (const auto& well: schedule_.getWells(reportStepNum)) {
+
+        // don't bother with wells not on this process
+        if (!wellQuery_(well.name())) {
+            continue;
+        }
+
+        //add data infrastructure for shut wells
+        if (!wellDatas.count(well.name())) {
+            auto& wellData = wellDatas[well.name()];
+            wellData.connections.reserve(well.getConnections().size());
+            std::transform(well.getConnections().begin(),
+                           well.getConnections().end(),
+                           std::back_inserter(wellData.connections),
+                           [](const auto& connection)
+                           {
+                               data::Connection res;
+                               res.index = connection.global_index();
+                               return res;
+                           });
+            wellDatas.emplace(std::make_pair(well.name(), wellData));
+        }
+
+        data::Well& wellData = wellDatas.at(well.name());
+
+        auto cond_assign = [](double& dest, unsigned idx, const auto& map)
+        {
+            auto it = map.find(idx);
+            if (it != map.end()) {
+                dest = it->second;
+            }
+        };
+
+        std::for_each(wellData.connections.begin(), wellData.connections.end(),
+                      [&cond_assign, this](auto& connectionData)
+                      {
+                          const auto index = connectionData.index;
+                          cond_assign(connectionData.cell_pressure, index,
+                                      oilConnectionPressures_);
+                          cond_assign(connectionData.cell_saturation_water, index,
+                                      waterConnectionSaturations_);
+                          cond_assign(connectionData.cell_saturation_gas, index,
+                                      gasConnectionSaturations_);
+                       });
+    }
+
+    oilConnectionPressures_.clear();
+    waterConnectionSaturations_.clear();
+    gasConnectionSaturations_.clear();
+}
+
+template<class FluidSystem>
+void RFTContainer<FluidSystem>::
+allocate(const std::size_t reportStepNum)
+{
+    // Well RFT data
+    const auto& rft_config = schedule_[reportStepNum].rft_config();
+    for (const auto& well: schedule_.getWells(reportStepNum)) {
+
+        // don't bother with wells not on this process
+        if (!wellQuery_(well.name())) {
+            continue;
+        }
+
+        if (!rft_config.active()) {
+            continue;
+        }
```

Comment:
This `!active` check could just go ahead of the loop too.

Reply:
done and done. sorry for being sloppy..